### ISSUE TITLE
How to get all thumb sizes?

### ISSUE THREAD

**Issue by joaoeduardo** [jgrossi/corcel#407]
- Corcel 2.5
- Laravel 5.5
- PHP 7.0 
- MySQL 5.7

How can I get all thumb sizes from a post?

I tried the code below, but I only get the original image:
```blade
{{ dump($post->thumbnail) }}
```

**Comment by fabiotnt** [comment]
You can get with:

`$post->attachment[0]->meta[1]->value;`

But you need to validate if ->image is not null before.

**Comment by fabiotnt** [comment]
Another approach:

`preg_match('/\.[^\.]+$/i',$post->image,$ext);
                $post->image_thumb  = str_replace($ext[0], "-150x150".$ext[0], $post->image);
                $post->image_medium = str_replace($ext[0], "-300x158".$ext[0], $post->image);
                $post->image_medium_large = str_replace($ext[0], "-768x403".$ext[0], $post->image);`

**Comment by jgrossi** [comment]
@joaoeduardo @fabiotnt you can get it through the `thumbnail()` relation, which returns a `ThumbnailMeta` instance: 
```php
$post->thumbnail->size(ThumbnailMeta::SIZE_LARGE);
```
take a look on the following test case: `ThumbnailTest::it_has_different_sizes`. and also the `Post` class on the `thumbnail()` relation (returning a `ThumbnailMeta` instance).

@joaoeduardo you are getting the original image because the `ThumbnailMeta` class has a `__toString()` method so if you just echo it's a string, otherwise, it's an object you have the `size()` method. 😉

let me know when I can close this issue, please 💪

**Comment by jgrossi** [comment]
@joaoeduardo @fabiotnt I just noticed that this info is not in the `README` file. if any of you can send a PR including that info, I appreciate 🙏

**Comment by joaoeduardo** [comment]
I will try. Seems simple. After I will send a pull request to `README` file.

**Comment by jgrossi** [comment]
thank you @joaoeduardo. once you do that please include `fixes #407` on PR description to make sure this issue is closed right after the PR is merged 😎
